Snakefile

```diff
@@ -7,6 +7,81 @@ wildcard_constraints:
 # defined before extra rules `include`d as they reference this constant
 SEGMENTS = ["pb2", "pb1", "pa", "ha","np", "na", "mp", "ns"]
 
+# ----------------------------------------------------------------------------
+# Allow this to work from a separate workdir by using a config in that workdir
+# which extends one of our base configs
+# ----------------------------------------------------------------------------
+if os.path.exists("config.yaml"):
+    configfile: "config.yaml"
+    # See commentary below
+    # print("This doesn't work as expected! See commentary in Snakefile", file=sys.stderr)
+    # exit(2)
+
+if config.get('extends', False):
+    extend_path = os.path.join(workflow.basedir, "config", config['extends'])
+    if not os.path.isfile(extend_path):
+        sys.exit(f"Your config tried to extend {config['extends']!r} but this doesn't exist. It must be relative to {os.path.join(workflow.basedir, 'config')}")
+    configfile: extend_path
+
+# NOTE:
+# In the situation where we're running outside of the repo, and we have a custom config YAML
+# such as `foo.yaml`:
+#        extends: h5n1-cattle-outbreak.yaml
+#        segments: ['pb2']
+# If we run with `--configfile foo.yaml` then the merging behaviour is strange,
+# as we merge in all the config values of `h5n1-cattle-outbreak.yaml` (via the above code)
+# which you would expect would lead to a config with the 9 segments defined in
+# 'h5n1-cattle-outbreak.yaml', however we end up with only config['segments']=['pb2'].
+# So it seems like the `--configfile` definitions are being re-applied a second time?!?
+#
+# This _is not the case_ when we use the `os.path.exists("config.yaml")` approach,
+# which is why it's not going to work without the following additional update_config
+# step (or something else?)
+
+if os.path.exists("config.yaml"):
+    # Following <https://github.com/snakemake/snakemake/blob/76d53290a003891c5ee41f81e8eb4821c406255d/snakemake/common/configfile.py#L7-L33>
+    import yte
+    with open("config.yaml", encoding='utf-8') as f:
+        overwrite_config = yte.process_yaml(f, require_use_yte=True)
+    snakemake.utils.update_config(config, overwrite_config)
```

Comment:
@tsibley Have you prototyped how you'd have this work for workflows such as these which don't have the `defaults/config.yaml` workflow config that measles does?

Reply:
I'm a bit unsure I'm correctly following what the above code is trying to do or what issue the comment is describing. Could you restate it?

My best guess at the missing bit of information is this:

> So it seems like the `--configfile` definitions are being re-applied a second time?!?

Snakemake merges config in this fixed order:

    configfile: < --configfile < --config

so I think maybe what you're seeing and perplexed by is that even though you're conditionally declaring `configfile: extend_path` based on a value from the already-loaded `--configfile`, the former has less precedence than the latter.

> workflows such as these which don't have the `defaults/config.yaml` workflow config that measles does?

I'd expect this to be equivalent to a repo like measles with an empty `defaults/config.yaml`, e.g. `{}`. But again, not sure I'm fully grokking what the above code is trying to accomplish, particularly re: `extends:`.

Reply:
You may see some Snakemake output near the start of the run like:

> Config file h5n1-cattle-outbreak.yaml is extended by additional config specified via the command line.

where the "additional config" is `--configfile foo.yaml`.

Reply:
> A workflow which expected to be run with `--configfile`, i.e. there's no "default configuration values". This is how mpox works too, among others.

Ohhh, reading the rest of this PR now, I think I get it. I'll respond more later.

Reply:
I updated the comment (via [a force-push yesterday](https://github.com/nextstrain/avian-flu/compare/b6e3bf784d85bd0461b0a139b8100441767353c4..e6b070d9417975f6d0415a0a4a65b5bf72ce51ca)) which may make things clearer?

Reply:
> Snakemake merges config in this fixed order:

> `configfile: < --configfile < --config`

> so I think maybe what you're seeing and perplexed by is that even though you're conditionally declaring configfile: extend_path based on a value from the already-loaded --configfile, the former has less precedence than the latter.

Yeah, and in most use cases how this works is very intuitive, but here the implementation seems subtle. Using the example from comment in the code with `--configfile foo.yaml` we have the following steps:
1. `--configfile` is parsed and set to `config`. `config['segments'] = ['pb2']`. No issues here. 
2. `config['extends']` us used to set the overlay via `configfile: extend_path` (which I refer to as the base config) 
3. My hunch was that now `config['segments'] = ['pb2', 'pb1', ...]` as that's what the base config defines, however it's not, it's still `config['segments'] = ['pb2']`

Note that if (in Snakemake code) you add between steps (1) and (2) `config['segments'] = 'whatever'` then that doesn't change the observations at step 3. This indicates to me that within step (2) snakemake actually merges the base config into `config` and then re-merges in `foo.yaml` again in order to enforce the precedence